dandi/tests/fixtures.py

```diff
@@ -221,8 +233,56 @@ def fixture() -> Iterator[str]:
     return fixture
 
 
+def get_filtered_gitrepo_fixture(
+    url: str,
+    whitelist: List[str],
+) -> Callable[[], Iterator[str]]:
+    @pytest.fixture(scope="session")
+    def fixture() -> Iterator[str]:
+        skipif.no_network()
+        skipif.no_git()
+        path = tempfile.mktemp()  # not using pytest's tmpdir fixture to not
+        # collide in different scopes etc. But we
```

Comment:
Please explain this "to not collide in different scopes" thing.  Is the issue that `tmp_path` is not available when `scope="session"`?  That's when you should use [`tmp_path_factory`](https://docs.pytest.org/en/7.1.x/reference/reference.html#std-fixture-tmp_path_factory).  Is the issue that you think calling `tmp_path_factory.mktemp()` with the same arguments will return the same path?  It doesn't; each call by default creates a new directory suffixed with an incrementing counter.

Reply:
Hm, ok, I tried to use tmp_path_factory, but it seems to have trouble dealing with windows:

```
2022-09-20T18:21:58.4183372Z ____________ ERROR at setup of test_upload_bids_validation_ignore _____________
2022-09-20T18:21:58.4183687Z dandi\tests\fixtures.py:259: in fixture
2022-09-20T18:21:58.4183940Z     capture_output=True,
2022-09-20T18:21:58.4184272Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\subprocess.py:488: in run
2022-09-20T18:21:58.4184625Z     with Popen(*popenargs, **kwargs) as process:
2022-09-20T18:21:58.4184993Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\subprocess.py:800: in __init__
2022-09-20T18:21:58.4185328Z     restore_signals, start_new_session)
2022-09-20T18:21:58.4185690Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\subprocess.py:1148: in _execute_child
2022-09-20T18:21:58.4186216Z     args = list2cmdline(args)
2022-09-20T18:21:58.4186549Z C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\subprocess.py:555: in list2cmdline
2022-09-20T18:21:58.4186915Z     needquote = (" " in arg) or ("\t" in arg) or not arg
2022-09-20T18:21:58.4187284Z E   TypeError: argument of type 'WindowsPath' is not iterable
```

As for the original code and comment, I used it to make the function analogous to `get_gitrepo_fixture()` — it's basically the same thing other than doing some git magic to really only ever download the whitelist. I think the original code might be from @yarikoptic

Reply:
I believe the problem is that `tmp_path_factory` returns a `pathlib.Path` object, yet `subprocess.run()` does not accept `Path`s in command arguments prior to Python 3.8.  You need to change `path` in the "clone" call to `str(path)`.  (Also, get rid of the `try` ... `finally` and its cleanup code, as pytest manages its temp dirs itself.)